sky/serve/autoscalers.py

```diff
@@ -132,93 +131,116 @@ def collect_request_information(
-            target_num_replicas = int(scale_target * num_replicas)
-        elif (self.lower_threshold is not None and
-              requests_per_replica < self.lower_threshold):
-            scale_target = requests_per_replica / self.lower_threshold
-            target_num_replicas = int(scale_target * num_replicas)
-
+        target_num_replicas = math.ceil(num_requests_per_second /
+                                        self.target_qps_per_replica)
         target_num_replicas = max(self.min_replicas,
                                   min(self.max_replicas, target_num_replicas))
+        logger.info(f'Requests per second: {num_requests_per_second}, '
+                    f'Current target number of replicas: {target_num_replicas}')
 
-        logger.info(f'Target number of replicas: {target_num_replicas}, '
-                    f'min_replicas: {self.min_replicas}, '
-                    f'max_replicas: {self.max_replicas}')
-
-        num_replicas_delta = 0
-        if num_replicas < self.min_replicas:
-            num_replicas_delta = self.min_replicas - num_replicas
-            self.upscale_counter = 0
-        elif num_replicas > self.max_replicas:
-            num_replicas_delta = self.max_replicas - num_replicas
-            self.downscale_counter = 0
-        elif target_num_replicas > num_replicas:
+        if target_num_replicas > self.target_num_replicas:
             self.upscale_counter += 1
             self.downscale_counter = 0
             if self.upscale_counter >= self.scale_up_consecutive_periods:
                 self.upscale_counter = 0
-                num_replicas_delta = target_num_replicas - num_replicas
-        elif target_num_replicas < num_replicas:
+                return target_num_replicas
+        elif target_num_replicas < self.target_num_replicas:
             self.downscale_counter += 1
             self.upscale_counter = 0
             if self.downscale_counter >= self.scale_down_consecutive_periods:
                 self.downscale_counter = 0
-                num_replicas_delta = target_num_replicas - num_replicas
+                return target_num_replicas
         else:
             self.upscale_counter = self.downscale_counter = 0
-        logger.info(f'Upscale counter: {self.upscale_counter}/'
-                    f'{self.scale_up_consecutive_periods}. '
-                    f'Downscale counter: {self.downscale_counter}/'
-                    f'{self.scale_down_consecutive_periods}')
+        return self.target_num_replicas
+
+    def _get_spot_resources_override_dict(self) -> Dict[str, Any]:
+        return {'use_spot': True, 'spot_recovery': None}
+
+    def _get_on_demand_resources_override_dict(self) -> Dict[str, Any]:
+        return {'use_spot': False, 'spot_recovery': None}
+
+    def _get_resources_override_dict(self, use_spot: bool) -> Dict[str, Any]:
+        return (self._get_spot_resources_override_dict()
+                if use_spot else self._get_on_demand_resources_override_dict())
```

Comment:
Do we really need these in this PR?

Reply:
I think we might want to include them for future.
if we want to include `resource_override_dict`, it is used for differentiating use_spot or not use_spot
@cblmemo

Reply:
from offline discussion, we set it to be None